### PR TITLE
PTX-4732 Automate node crash

### PR DESCRIPTION
**Pull request by aravindranath14**
**What this PR does / why we need it**:
Automate Node crash
Crash each node with volume one by one

**Which issue(s) this PR fixes** (optional)
Closes #PTX-4732

**Special notes for your reviewer**:

### CODE REVIEW

File: tests/nodeCrash/nodeCrash_test.go [hunk 1]
```diff
@@ -0,0 +1,133 @@
+package tests
+
+import (
```
Comment (lsrinivas-pure) [inline tests/nodeCrash/nodeCrash_test.go:3]: this should be part of longevity, please add the test in testTriggers

**Review by lsrinivas-pure — CHANGES_REQUESTED** [review]

File: tests/nodeCrash/nodeCrash_test.go (hunk 1 above)
Comment (aravindranath14) [reply tests/nodeCrash/nodeCrash_test.go:3]: Thanks for the comments. I have added the test in testTriggers

Comment (lsrinivas-pure) [reply tests/nodeCrash/nodeCrash_test.go:3]: u can delete this

File: tests/testTriggers.go [hunk 2]
```diff
@@ -654,6 +656,79 @@ func TriggerRebootNodes(contexts *[]*scheduler.Context, recordChan *chan *EventR
...
+		Step(fmt.Sprintf("crash node one at a time from the node(s): %v", nodesToCrash), func() {
+			// TODO: Below is the same code from existing nodeCrash test
+			for _, n := range nodesToCrash {
+				if n.IsStorageDriverInstalled {
+					Step(fmt.Sprintf("crash node: %s", n.Name), func() {
+						taskStep := fmt.Sprintf("crash node: %s.", n.MgmtIp)
+						event.Event.Type += "<br>" + taskStep
+						err := Inst().N.CrashNode(n, node.CrashNodeOpts{
+							Force: true,
+							ConnectionOpts: node.ConnectionOpts{
```
Comment (lsrinivas-pure) [inline tests/testTriggers.go:689]: please make necessary changes in longevity_test.go to trigger this test.
Also attach the successful run result

**Review by lsrinivas-pure — CHANGES_REQUESTED** [review]

File: tests/testTriggers.go [hunk 3]
```diff
@@ -654,6 +656,79 @@ func TriggerRebootNodes(contexts *[]*scheduler.Context, recordChan *chan *EventR
...
+		Outcome: []error{},
+	}
+
+	defer func() {
+		event.End = time.Now().Format(time.RFC1123)
+		*recordChan <- event
+	}()
+
+	Step("get all nodes and crash one by one", func() {
+		nodesToCrash := node.GetWorkerNodes()
```
Comment (mborodin-px) [inline tests/testTriggers.go:677]: the definition of the test says `crashes node on which apps are running` but it crashes all worker nodes

File: drivers/node/node.go [hunk 4]
```diff
@@ -68,6 +68,12 @@ type RebootNodeOpts struct {
 	ConnectionOpts
 }
 
+// CrashNodeOpts provide additional options for crash operation
+type CrashNodeOpts struct {
+	Force bool
```
[the remark this answers is not in the document]
Comment (aravindranath14) [reply drivers/node/node.go:73]: This is used in nodeCrash_test.go, line: 68

File: tests/testTriggers.go (hunk 2 above)
Comment (aravindranath14) [reply tests/testTriggers.go:699]: made necessary changes to longevity_test.go; Please find the attached run lo
[longevity.log](https://github.com/portworx/torpedo/files/7888457/longevity.log)
g

File: tests/nodeCrash/nodeCrash_test.go [hunk 5]
```diff
@@ -0,0 +1,133 @@
...
+								Timeout:         defaultTestConnectionTimeout,
+								TimeBeforeRetry: defaultWaitRebootRetry,
+							})
+							Expect(err).NotTo(HaveOccurred())
+						})
+
+						//Step(fmt.Sprintf("wait for volume driver to stop on node: %v", n.Name), func() {
+						//	err := Inst().V.WaitDriverDownOnNode(n)
+						//	Expect(err).NotTo(HaveOccurred())
+						//})
```
[the remark this answers is not in the document]
Comment (aravindranath14) [reply tests/nodeCrash/nodeCrash_test.go]: removed

File: tests/testTriggers.go [hunk 6]
```diff
@@ -149,6 +149,8 @@ const (
 	CrashVolDriver = "crashVolDriver"
 	// RebootNode reboots alll nodes one by one
 	RebootNode = "rebootNode"
+	// CrashNode crashes alll nodes one by one
```
[the remark this answers is not in the document]
Comment (aravindranath14) [reply tests/testTriggers.go]: Done

File: tests/testTriggers.go (hunk 3 above)
Comment (aravindranath14) [reply tests/testTriggers.go:687]: Modified the definintion

**Comment by thiguetta** [comment]
@aravindranath14 have we tested it somewhere? please post the job link

**Comment by aravindranath14** [comment]
Here is a log file captured for successful longevity run of crashnode test:
(multiple iterations ran overnight. Manually stopped)
[longevity.log](https://github.com/portworx/torpedo/files/7902668/longevity.log)

**Comment by lsrinivas-pure** [comment]
LGTM.
run this job once using docker build with your changes
https://jenkins.portworx.dev/job/Torpedo/job/tp-dev-test/

**Comment by aravindranath14** [comment]
> 

Here is a successful run from the branch

File: tests/longevity/longevity_test.go [hunk 7]
```diff
@@ -447,6 +450,17 @@ func populateIntervals() {
...
 
+	triggerInterval[CrashNode][10] = 1 * baseInterval
+	triggerInterval[CrashNode][9] = 3 * baseInterval
+	triggerInterval[CrashNode][8] = 6 * baseInterval
+	triggerInterval[CrashNode][7] = 9 * baseInterval
+	triggerInterval[CrashNode][6] = 12 * baseInterval
+	triggerInterval[CrashNode][5] = 15 * baseInterval
+	triggerInterval[CrashNode][4] = 18 * baseInterval
+	triggerInterval[CrashNode][3] = 21 * baseInterval
+	triggerInterval[CrashNode][2] = 24 * baseInterval
```
Comment (lsrinivas-pure) [inline tests/longevity/longevity_test.go:461]: can you please add triggerinterval 0 for crashnode. it is used to disable the test if needed

**Review by lsrinivas-pure — APPROVED** [review]
LGTM

**Pull request merged**